velox/functions/prestosql/json/JsonStringUtil.cpp

```diff
@@ -0,0 +1,194 @@
+    return 4;
+  }
+  return -1;
+}
+
+FOLLY_ALWAYS_INLINE char hexDigit(uint8_t c) {
+  return c < 10 ? c + '0' : c - 10 + 'A';
+}
+
+FOLLY_ALWAYS_INLINE void writeHex(char16_t value, char*& out) {
+  value = folly::Endian::little(value);
+  *out++ = '\\';
+  *out++ = 'u';
+  *out++ = hexDigit((value >> 12) & 0x0F);
+  *out++ = hexDigit((value >> 8) & 0x0F);
+  *out++ = hexDigit((value >> 4) & 0x0F);
+  *out++ = hexDigit(value & 0x0F);
+  return;
+}
+
+std::array<int8_t, 128> getAsciiEscapes() {
+  std::array<int8_t, 128> escapes;
+  std::fill(escapes.data(), escapes.data() + 32, -1);
+  escapes['"'] = '"';
+  escapes['\\'] = '\\';
+  escapes['\b'] = 'b';
+  escapes['\t'] = 't';
+  escapes['\n'] = 'n';
+  escapes['\f'] = 'f';
+  escapes['\r'] = 'r';
+  return escapes;
+}
+static const std::array<int8_t, 128> asciiEscapes = getAsciiEscapes();
+
+FOLLY_ALWAYS_INLINE void encodeAscii(int8_t value, char*& out) {
+  int8_t escapeCode = asciiEscapes[value];
+  if (escapeCode == 0) {
+    *out++ = char(value);
+  } else if (escapeCode > 0) {
+    *out++ = '\\';
+    *out++ = char(escapeCode);
+  } else {
+    writeHex(value, out);
+  }
+  return;
+}
+
+FOLLY_ALWAYS_INLINE size_t encodedAsciiSize(uint8_t value) {
+  int8_t escapeCode = asciiEscapes[value];
+  if (escapeCode == 0) {
+    return 1;
+  }
+  if (escapeCode > 0) {
+    return 2;
+  }
+  return 6;
+}
+
+FOLLY_ALWAYS_INLINE void encodeUtf16Hex(char32_t codePoint, char*& out) {
+  VELOX_DCHECK(codePoint <= 0x10FFFFu);
+  if (codePoint >= 0x10000u) {
+    writeHex(
+        static_cast<char16_t>(
+            0xD800u + (((codePoint - 0x10000u) >> 10) & 0x3FFu)),
+        out);
+    writeHex(
+        static_cast<char16_t>(0xDC00u + ((codePoint - 0x10000u) & 0x3FFu)),
+        out);
+    return;
+  }
+  writeHex(static_cast<char16_t>(codePoint), out);
+  return;
+}
+
+} // namespace
+
+void encodeUtf16HexForTest(char32_t codePoint, char*& out) {
+  encodeUtf16Hex(codePoint, out);
+}
+
+void escapeString(const char* input, size_t length, char* output) {
+  char* pos = output;
+
+  *pos++ = '"';
+
+  auto* start = reinterpret_cast<const unsigned char*>(input);
+  auto* end = reinterpret_cast<const unsigned char*>(input + length);
+  while (start < end) {
+    int count = validateAndGetNextUtf8Length(start, end);
+    if (count == -1) {
+      writeHex(0xFFFDu, pos);
+      start++;
+      continue;
+    }
+    if (count == 1) {
+      encodeAscii(int8_t(*start), pos);
+      start++;
+      continue;
+    }
+    if (count == 2 || count == 3) {
+      for (int i = 0; i < count; i++) {
+        *pos++ = char(*start++);
+      }
+      continue;
+    }
+    char32_t codePoint = folly::utf8ToCodePoint(start, end, true);
+    if (codePoint == U'\ufffd') {
+      writeHex(0xFFFDu, pos);
+      continue;
+    }
+    encodeUtf16Hex(codePoint, pos);
+  }
+
+  *pos++ = '"';
+}
+
+size_t escapedStringSize(const char* input, size_t length) {
+  const size_t ENCODED_HEX_SIZE = 6;
```

Comment:
Convention is `constexpr size_t kEncodedHexSize = 6`

Reply:
Revised

Reply:
Looks like it's still not updated?

Reply:
Updated. Failed to upload the updated version somehow.

Reply:
It's still not updated